### PR TITLE
Fix error msgs related to initial condition + add tests for ValueErrors

### PR DESCRIPTION
**Pull request by navidcy**

### CODE REVIEW

File: regional_mom6/regional_mom6.py [hunk 1]
```diff
@@ -592,28 +592,29 @@ def initial_condition(
...
-        # Separate out tracers from two velocity fields of IC
+        # Separate out IC's tracers, two velocities, and free surface
         try:
             ic_raw_tracers = ic_raw[
                 [varnames["tracers"][i] for i in varnames["tracers"]]
             ]
         except:
             raise ValueError(
-                "Error in reading in initial condition tracers. Terminating!"
+                "Error while reading initial conditions tracers. Terminating!"
```
Comment (angus-g) [inline regional_mom6/regional_mom6.py:602]:
```suggestion
                "Error while reading initial condition tracers. Terminating!"
```

File: codecov.yml [hunk 2]
```diff
@@ -6,6 +6,7 @@ codecov:
 comment: false
 
 ignore:
+  - "3/*"
```
Comment (angus-g) [inline codecov.yml:9]: What is this?

File: tests/test_expt_class.py [hunk 3]
```diff
@@ -160,80 +160,132 @@ def test_ocean_forcing(
...
+        np.random.random((100, 100, 10)),
+        dims=dims,
+        coords=coords,
+    )
+
+    v = xr.DataArray(
+        np.random.random((100, 100, 10)),
+        dims=dims,
+        coords=coords,
+    )
```
Comment (angus-g) [inline tests/test_expt_class.py:201]: Probably clearer with something like:

```python
def rand_ds():
   return xr.DataArray(
        np.random.random((100, 100, 10)),
        dims=dims,
        coords=coords,
    )

ic_datasets = {var: rand_ds() for var in ["temp", "salt", "u", "v"]}
```

File: tests/test_expt_class.py [hunk 4]
```diff
@@ -160,80 +160,132 @@ def test_ocean_forcing(
...
+            "v": v,
+        }
+    )
+
+    initial_cond_without_v = xr.Dataset(
+        {
+            "temp": temp,
+            "salt": salt,
+            "eta": eta,
+            "u": u,
```
Comment (angus-g) [inline tests/test_expt_class.py:265]: Maybe like...
```python
def ic_dataset_without(ics, keys):
    return xr.Dataset(
        {k: v for k, v in ics.items() if k not in keys}
    )
```

**Review by angus-g — COMMENTED** [review]
I think there are a few tweaks to the test to reduce duplication/increase clarity, but I do like the extended coverage by adding all the new conditions!

File: codecov.yml (hunk 2 above)
Comment (navidcy) [reply codecov.yml:9]: Sorry, I should have explained.
That's my attempt to remove this `3` directory from the codecov report, e.g., see 

Screenshot 2024-04-26 at 07 16 12

I have no idea why is there! Does it come from a docker/CI on the notebook?

File: tests/test_expt_class.py (hunk 4 above)
Comment (navidcy) [reply tests/test_expt_class.py:265]: Thanks for the suggestions! I would really like to reduce the boilerplate code but I didn't have the python-expertise to do it from the top of my head!

**Comment by navidcy** [comment]
this is WIP -- I think I know how to proceed just didn't find the time!

**Comment by aekiss** [comment]
maybe convert to a draft PR to avoid confusion?

**Comment by navidcy** [comment]
great idea!

**Comment by navidcy** [comment]
I'm closing this as stale :(

**Pull request closed without merging**